src/content/SectionPopup/LocationsComponent/LocationComponent.tsx

```diff
@@ -1,42 +1,30 @@
 import "../PopupComponent.css"
 
 interface IProps {
-  locations: string[]
+  locations: Set<{ location: string | undefined; link: string } | undefined>
```

Comment:
This is very strange interface. Why not String[] and do the link formation in this component?

Reply:
yeah thats what I did initially, and then @ThatNerdSquared suggested that parsing be done in the parent and that locations component only deal with rendering the locations. Im fine with either approach, the commits before the review have the approach you mention